src/main/java/frc/robot/vision/multivisionsources/MultiVisionSources.java

```diff
@@ -0,0 +1,78 @@
+package frc.robot.vision.multivisionsources;
+
+import frc.robot.subsystems.GBSubsystem;
+import frc.robot.vision.VisionConstants;
+import frc.robot.vision.rawdata.RawVisionData;
+import frc.robot.vision.sources.VisionSource;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class MultiVisionSources<ReturnType extends RawVisionData> extends GBSubsystem {
+
+	private final List<VisionSource<ReturnType>> visionSources;
+
+	@SafeVarargs
+	public MultiVisionSources(String logPath, VisionSource<ReturnType>... visionSources) {
+		super(logPath);
+		this.visionSources = List.of(visionSources);
+	}
+
+	public MultiVisionSources(String logPath, List<VisionSource<ReturnType>> visionSources) {
+		super(logPath);
+		this.visionSources = visionSources;
+	}
+
+	protected List<VisionSource<ReturnType>> getVisionSources() {
+		return visionSources;
+	}
+
+	protected ArrayList<ReturnType> createMappedCopyOfSources(
+		List<VisionSource<ReturnType>> list,
+		Function<Optional<ReturnType>, Optional<ReturnType>> mapping
+	) {
+		ArrayList<ReturnType> output = new ArrayList<>();
+		list.forEach(visionSource -> {
+			visionSource.update();
+			Optional<ReturnType> observation = mapping.apply(visionSource.getRawVisionData());
+			observation.ifPresent(output::add);
+		});
+		return output;
+	}
```

Comment:
not a good name, it does not create a mapped copy of the sources

Reply:
mapping is a bijection two, not necessarily equal, sets.